pkg/core/children.go

```diff
@@ -99,10 +115,40 @@ func getChildNamesByType(obj *appsv1.Deployment) (map[string]struct{}, map[strin
 	for _, container := range obj.Spec.Template.Spec.Containers {
 		for _, env := range container.EnvFrom {
 			if cm := env.ConfigMapRef; cm != nil {
-				configMaps[cm.Name] = struct{}{}
+				configMaps[cm.Name] = &configMetadata{required: true, allKeys: true, keys: map[string]struct{}{}}
 			}
 			if s := env.SecretRef; s != nil {
-				secrets[s.Name] = struct{}{}
+				secrets[s.Name] = &configMetadata{required: true, allKeys: true, keys: map[string]struct{}{}}
+			}
+		}
+	}
+
+	// Range through all Containers and their respective Env
+	for _, container := range obj.Spec.Template.Spec.Containers {
+		for _, env := range container.Env {
+			if valFrom := env.ValueFrom; valFrom != nil {
+				if cm := valFrom.ConfigMapKeyRef; cm != nil {
```

Comment:
I think we can make this nicer to read and remove the need for `*configMetadata` pointers everywhere if we add a function

```
func parseConfigMapKeyRef(meta configMetadata, cm *corev1.ConfigMapKeySelector) configMetadata {
	if !meta.allKeys {
		if meta.keys == nil {
			meta.keys = make(map[string]struct{})
		}
		if cm.Optional == nil || !*cm.Optional {
			meta.required = true
		}
		meta.keys[cm.Key] = struct{}{}
	}
	return meta
}
```

and change the entire contents of this `if` with:

```
configMaps[cm.Name] = parseConfigMapKeyRef(configmaps[cm.Name], cm)
```

This will not only make this for loop have less nesting and be easier to read, but removing the need for pointers everywhere will help clean this up a bit too and we won't need to have the anonymous functions in `getCurrentChildren` either.

This would lay out the pattern required to add field parsing to the other mount types too which is an added bonus!

We can do the same for the `SecretKeyRef` too

Reply:
Done